src/generators/java/jsoup.ts

```diff
@@ -0,0 +1,159 @@
+  javaCode += "\nclass Main {\n\n";
+
+  javaCode += "\tpublic static void main(String[] args) throws IOException {\n";
+  javaCode +=
+    '\t\tConnection.Response response = Jsoup.connect("' + request.url + '")\n';
+
+  let gzip = false;
+  if (request.headers) {
+    for (const [headerName, headerValue] of request.headers) {
+      if (headerValue === null) {
+        continue;
+      }
+
+      if (headerName.toLowerCase() === "user-agent") {
+        javaCode +=
+          "\t\t.userAgent" + '("' + doubleQuotes(headerValue) + '")\n';
+      } else if (headerName.toLowerCase() === "cookie") {
+        const cookieValues = headerValue.split(";");
+        for (const index in cookieValues) {
+          javaCode +=
+            '\t\t.cookie("' +
+            cookieValues[index].split("=")[0].trim() +
+            '", "' +
+            cookieValues[index].split("=")[1].trim() +
+            '")\n';
+        }
+      } else {
+        javaCode +=
+          '\t\t.header("' +
+          headerName +
+          '", "' +
+          doubleQuotes(headerValue) +
+          '")\n';
+      }
+
+      if (headerName.toLowerCase() === "accept-encoding" && headerValue) {
+        gzip = headerValue.indexOf("gzip") !== -1;
+      }
+    }
+  }
+
+  if (request.auth) {
+    javaCode +=
+      '\t\tbyte[] message = ("' +
+      doubleQuotes(request.auth.join(":")) +
+      '").getBytes("UTF-8");\n';
+    javaCode +=
+      "\t\tString basicAuth = DatatypeConverter.printBase64Binary(message)\n";
+    javaCode += '\t\t.header("Authorization", "Basic " + basicAuth)\n';
+    javaCode += "\n";
+  }
+
+  if (request.data) {
+    request.data = doubleQuotes(request.data);
+
+    javaCode += '\t\t.requestBody("' + request.data + '")\n';
+  }
+
+  if (request.multipartUploads) {
+    javaCode += '\t\t.data("';
+
+    for (const m of request.multipartUploads) {
+      const name = m.name;
+      const fileInputStream = "new FileInputStream(new File(";
+      const fileName = "filename";
+      if ("contentFile" in m) {
+        if (m.contentFile === "-") {
+          if (request.stdinFile) {
+            javaCode +=
+              name +
+              '", "' +
+              fileName +
+              '", ' +
+              fileInputStream +
+              request.stdinFile;
+          } else if (request.stdin) {
+            javaCode +=
+              name +
+              '", "' +
+              fileName +
+              '", ' +
+              fileInputStream +
+              request.stdin;
+          }
+        } else {
+          javaCode +=
+            name + '", "' + fileName + '", ' + fileInputStream + m.contentFile;
+        }
+      } else {
+        javaCode +=
+          name + '", "' + fileName + '",' + fileInputStream + m.content;
+      }
+    }
+    javaCode += ")))\n";
+  }
+
+  javaCode +=
+    "\t\t.method(org.jsoup.Connection.Method." + request.method + ")\n";
+  javaCode += "\t\t.ignoreContentType(true)\n";
+  javaCode += "\t\t.timeout(30000)\n";
```

Comment:
This should come from request.timeout and not be added otherwise if it doesn't exist.